storage/storage.go

```diff
@@ -246,13 +176,21 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 // You may configure the client by passing in options from the [google.golang.org/api/option]
 // package.
 func NewGRPCClient(ctx context.Context, opts ...option.ClientOption) (*Client, error) {
+	var creds *auth.Credentials
+
+	c, err := internaloption.AuthCreds(ctx, opts)
+	if err == nil {
+		creds = c
+	}
+
```

Comment:
Can we extract this to a separate function used for both http and grpc? Also, I think we need to add default options before doing this.